hedera-node/hedera-app/src/test/java/com/hedera/node/app/authorization/PrivilegesVerifierTest.java

```diff
@@ -0,0 +1,678 @@
+            com.hedera.hapi.node.base.AccountID accountID = PbjConverter.toPbj(accessor.getPayer());
+            com.hedera.hapi.node.base.HederaFunctionality functionality = PbjConverter.toPbj(accessor.getFunction());
+            com.hedera.hapi.node.transaction.TransactionBody txBody = PbjConverter.toPbj(accessor.getTxn());
+            final var pbjResult = delegate.hasPrivileges(accountID, functionality, txBody);
+            return SystemOpAuthorization.valueOf(pbjResult.name());
+        }
+
+        boolean canPerformNonCryptoUpdate(final long accountNum, final long fileNum) {
+            final var accountID = com.hedera.hapi.node.base.AccountID.newBuilder()
+                    .accountNum(accountNum)
+                    .build();
+            final var fileID = com.hedera.hapi.node.base.FileID.newBuilder()
+                    .fileNum(fileNum)
+                    .build();
+            final var fileUpdateTxBody = com.hedera.hapi.node.transaction.TransactionBody.newBuilder()
+                    .fileUpdate(com.hedera.hapi.node.file.FileUpdateTransactionBody.newBuilder()
+                            .fileID(fileID)
+                            .build())
+                    .build();
+            final var fileAppendTxBody = com.hedera.hapi.node.transaction.TransactionBody.newBuilder()
+                    .fileAppend(com.hedera.hapi.node.file.FileAppendTransactionBody.newBuilder()
+                            .fileID(fileID)
+                            .build())
+                    .build();
+            return delegate.hasPrivileges(accountID, HederaFunctionality.FILE_UPDATE, fileUpdateTxBody)
+                            == SystemPrivilege.AUTHORIZED
+                    && delegate.hasPrivileges(accountID, HederaFunctionality.FILE_APPEND, fileAppendTxBody)
+                            == SystemPrivilege.AUTHORIZED;
+        }
+    }
+
+    @BeforeEach
+    void setUp() {
+        final var configuration = HederaTestConfigBuilder.createConfig();
+        final ConfigProvider configProvider = () -> new VersionedConfigImpl(configuration, 1L);
+
+        subject = new Wrapper(configProvider);
+    }
+
+    @SuppressWarnings("DataFlowIssue")
+    @Test
+    void testConstructorWithIllegalParameters() {
+        assertThatThrownBy(() -> new PrivilegesVerifier(null)).isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void treasuryCanUpdateAllNonAccountEntities() {
+        // expect:
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 101));
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 102));
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 111));
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 112));
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 121));
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 122));
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 123));
+        for (var num = 150; num <= 159; num++) {
```

Comment:
This would be easier to comprehend if we used something line `MIN_UPGRADE_FILE` and `MAX_UPGRADE_FILE` as constants for the numbers.  Even better would be to read those values from the config so we don't have to change this code if/when the config defaults change.